src/huggingface_hub/community.py

```diff
@@ -0,0 +1,328 @@
+"""
+Data structures to interact with discussions and pull requests on the Hub.
+
+See [the discussions and pull requests guide](https://huggingface.co/docs/hub/repositories-pull-requests-discussions)
+for more information on pull requests, discussions, and the community tab.
+"""
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import List, Optional
+
+from dateutil.parser import parse as parse_datetime
+
+
+@dataclass
+class Discussion:
+    """
+    A discussion or pull request on the Hub.
+
+    Attributes:
+        title (`str`):
+            The title of the discussion / pull request
+        status (`str`):
```

Comment:
Maybe we can use `Literal` here

* https://docs.python.org/3/library/typing.html?highlight=typing#typing.Literal
* https://stackoverflow.com/questions/60599469/typing-restrict-to-a-list-of-strings